src/layers/core/geojson-layer/geojson-layer.js

```diff
@@ -72,27 +72,32 @@ export default class GeoJsonLayer extends Layer {
     }
   }
 
-  _onHoverSublayer(info) {
-    this.state.pickInfos.push(info);
-  }
-
-  pick(opts) {
-    super.pick(opts);
+  _onPickSublayer(mode, info) {
+    const {pickInfos, subLayerCount} = this.state;
+    pickInfos.push(info);
 
-    const info = this.state.pickInfos.find(i => i.index >= 0);
+    if (pickInfos.length === subLayerCount) {
+      // all sublayers have been accounted for
+      let pickInfo = pickInfos.find(i => i.index >= 0) || pickInfos[0];
 
-    if (opts.mode === 'hover') {
-      this.state.pickInfos = [];
-    }
+      pickInfo = Object.assign({}, pickInfo, {
+        layer: this,
+        feature: get(pickInfo, 'object.feature') || pickInfo.object
+      });
 
-    if (!info) {
-      return;
+      switch (mode) {
```

Comment:
Nit: I'd probably drop the switch statement, create two methods that call a common method. An argument for this is that hover and click are separate methods everywhere else, this breaks that pattern. Also, as suggested below, the common code might be a candidate for moving up to the base layer.